src/map/clif.c

```diff
@@ -20601,6 +20605,193 @@ void clif_ui_action(struct map_session_data *sd, int32 UIType, int32 data)
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", i, config_filename);
+	return true;
+}
+
+bool clif_stylist_read_db_libconfig_sub(struct config_setting_t *it, int idx, const char *source)
+{
+	struct stylist_data_entry entry = { 0 };
+	int i32 = 0, type = 0;
+	int64 i64 = 0;
+
+	nullpo_ret(it);
+	nullpo_ret(source);
+
+	if (!itemdb->lookup_const(it, "Type", &type) || type >= MAX_STYLIST_TYPE || type < 0) {
+		ShowWarning("clif_stylist_read_db_libconfig_sub: Invalid or missing Type (%d) in \"%s\", entry #%d, skipping.\n", type, source, idx);
+		return false;
+	}
+	if (!itemdb->lookup_const(it, "Id", &i32) || i32 <= 0) {
+		ShowWarning("clif_stylist_read_db_libconfig_sub: Invalid or missing Id (%d) in \"%s\", entry #%d, skipping.\n", i32, source, idx);
+		return false;
+	}
+	entry.id = i32;
+
+	if (libconfig->setting_lookup_int64(it, "Zeny", &i64)) {
+		if (i64 > MAX_ZENY) {
+			ShowWarning("clif_stylist_read_db_libconfig_sub: zeny is too big in \"%s\", entry #%d, capping to MAX_ZENY.\n", source, idx);
+			entry.zeny = MAX_ZENY;
+		} else {
+			entry.zeny = (int)i64;
+		}
+	}
+
+	if (itemdb->lookup_const(it, "ItemID", &i32))
+		entry.itemid = i32;
+
+	if (itemdb->lookup_const(it, "BoxItemID", &i32))
+		entry.boxid = i32;
+
+	VECTOR_ENSURE(stylist_data[type], 1, 1);
+	VECTOR_PUSH(stylist_data[type], entry);
+	return true;
+}
+
+bool clif_style_change_validate_requirements(struct map_session_data *sd, int type, uint16 idx)
+{
+	struct item it;
+	struct stylist_data_entry *entry;
+
+	nullpo_retr(false, sd);
+	Assert_retr(false, type >= 0 && type < MAX_STYLIST_TYPE);
+	Assert_retr(false, idx < VECTOR_LENGTH(stylist_data[type]));
+
+	entry = &VECTOR_INDEX(stylist_data[type], idx);
+
+	if (entry->id != 0) {
+		if (entry->zeny != 0) {
+			if (sd->status.zeny < entry->zeny)
+				return false;
+
+			sd->status.zeny -= entry->zeny;
+			clif->updatestatus(sd, SP_ZENY);
+		} else if (entry->itemid != 0) {
+			it.nameid = entry->itemid;
+			it.amount = 1;
+			return script->buildin_delitem_search(sd, &it, false);
+		} else if (entry->boxid != 0) {
+			it.nameid = entry->boxid;
+			it.amount = 1;
+			return script->buildin_delitem_search(sd, &it, false);
+		}
+		return true;
+	}
+	return false;
+}
+void clif_stylist_send_rodexitem(struct map_session_data *sd, int16 itemid)
+{
+	struct rodex_message msg = { 0 };
+
+	nullpo_retv(sd);
+
+	msg.receiver_id = sd->status.char_id;
+	msg.items[0].item.nameid = itemid;
+	msg.items[0].item.amount = 1;
+	msg.items[0].item.identify = 1;
+	msg.type = MAIL_TYPE_NPC | MAIL_TYPE_ITEM;
+
+	safestrncpy(msg.sender_name, msg_txt(366), NAME_LENGTH);
+	safestrncpy(msg.title, msg_txt(367), RODEX_TITLE_LENGTH);
+	safestrncpy(msg.body, msg_txt(368), MAIL_BODY_LENGTH);
+	msg.send_date = (int)time(NULL);
+	msg.expire_date = (int)time(NULL) + RODEX_EXPIRE;
+
+	intif->rodex_sendmail(&msg);
+}
+
+void clif_parse_cz_req_style_change(int fd, struct map_session_data *sd)
```

Comment:
missing nonnull attributes same like in other clif_parse_* functions.
Also next nullpo_retv(sd) useless if add nonnull attributes for sd parameter